drivers/include/sdcard_spi.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2016 Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_sdcard_spi SPI SD-Card driver
+ * @ingroup     drivers
+ * @brief       Driver for reading and writing sd-cards via spi interface.
+ * @{
+ *
+ * @file
+ * @brief       Public interface for the sdcard_spi driver.
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ */
+
+#ifndef SDCARD_SPI_H
+#define SDCARD_SPI_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#include "stdbool.h"
+#include "sdcard_spi_internal.h"
+
+#define SD_HC_BLOCK_SIZE 512
+
+#ifndef NUM_OF_SD_CARDS
+    #define NUM_OF_SD_CARDS 1
+#endif
```

Comment:
also see https://github.com/RIOT-OS/RIOT/wiki/Guide:-Writing-a-device-driver-in-RIOT for further reference